web/models.py

```
@@ -640,11 +643,69 @@ class CourseProgress(models.Model):
+                for attempt in final_attempts:
+                    score = attempt.calculate_score()
+                    if score >= final_exam.passing_score:
+                        passed_final = True
+                        break
+
+                if passed_final:
+                    completed_items += 1
+
+        except Exception:
+            # Fallback to just sessions if there's an error with exam calculations
+            pass
```

Ruff (0.8.2)

`try`-`except`-`pass` detected, consider logging the exception

(S110)